### PR TITLE
Feat missing files

### PR DESCRIPTION
**Pull request by JulienChampagnol**

### CODE REVIEW

File: components/ExtensionSelector.vue [hunk 1]
```diff
@@ -29,18 +32,29 @@
...
+  const loading = ref(true)
   const file_extensions = ref([])
 
+  const toggle_loading = useToggle(loading)
+
   async function get_output_file_extensions() {
     const params = new FormData()
     params.append("geode_object", geode_object)
+    loading.value = true
     await api_fetch(
```
Comment (BotellaA) [inline components/ExtensionSelector.vue:44]: If you await, you can just toggle back on the next instruction instead of in each interceptors

File: components/FileSelector.vue [hunk 2]
```diff
@@ -1,37 +1,69 @@
...
-    counter
-    show-size
-    @click:clear="stepper_tree.files = []"
-  />
+  <v-row>
+    <v-col class="pa-0">
+      <v-file-input
+        v-model="files"
+        :multiple="multiple"
+        :label="label"
```
Comment (BotellaA) [inline components/FileSelector.vue:7]: Is label really a FileSelector props or should it be hard coded?

File: components/FileSelector.vue [hunk 3]
```diff
@@ -53,7 +85,63 @@
...
+      accept.value = "*"
+    } else {
+      await get_allowed_files()
+    }
   })
 </script>
+
+<style scoped>
+  .v-btn {
+    text-transform: unset !important;
```
Comment (BotellaA) [inline components/FileSelector.vue:145]: not sure about that, most of the time I like uppercase

File: components/MissingFilesSelector.vue [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            },
+          },
+        )
+      }
+      reader.readAsDataURL(file)
+    }
+  }
+
+  onMounted(async () => {
+    await missing_files()
```
Comment (BotellaA) [inline components/MissingFilesSelector.vue:125]: this await will not do anything. Review your async/await calls.

File: components/ObjectSelector.vue [hunk 5]
```diff
@@ -39,17 +41,21 @@
...
     const params = new FormData()
     params.append("filename", files[0].name)
-
+    loading.value = true
     await api_fetch(
       `${route_prefix}/allowed_objects`,
       { method: "POST", body: params },
       {
         response_function: (response) => {
+          toggle_loading()
```
Comment (BotellaA) [inline components/ObjectSelector.vue:58]: What if there is an error?

File: components/FileSelector.vue [hunk 6]
```diff
@@ -1,37 +1,69 @@
...
   const stepper_tree = inject("stepper_tree")
   const { route_prefix } = stepper_tree
 
   const props = defineProps({
     multiple: { type: Boolean, required: true },
     label: { type: String, required: true },
-    variable_to_update: { type: String, required: true },
-    variable_to_increment: { type: String, required: true },
+    variable_to_update: { type: String, required: false },
+    variable_to_increment: { type: String, required: false },
```
Comment (BotellaA) [inline components/FileSelector.vue:43]: Why this change?

File: components/FileSelector.vue (hunk 2 above)
Comment (JulienChampagnol) [reply components/FileSelector.vue:7]: It changes depending if we have multiple true or false, maybe it should depend on that

File: components/FileSelector.vue (hunk 3 above)
Comment (JulienChampagnol) [reply components/FileSelector.vue:145]: It seems pretty strong to me, jut my opinion

File: components/ExtensionSelector.vue (hunk 1 above)
Comment (JulienChampagnol) [reply components/ExtensionSelector.vue:44]: True that

File: components/ObjectSelector.vue (hunk 5 above)
Comment (JulienChampagnol) [reply components/ObjectSelector.vue:58]: There's an error handler in api_fetch

File: components/FileSelector.vue (hunk 2 above)
Comment (BotellaA) [reply components/FileSelector.vue:7]: yes!

File: components/ObjectSelector.vue (hunk 5 above)
Comment (BotellaA) [reply components/ObjectSelector.vue:58]: yes, but the toggle is not toggle back

File: components/FileSelector.vue [hunk 7]
```diff
@@ -1,59 +1,59 @@
...
+    stepper_tree[variable_to_update] = value
+    if (value.length) {
+      stepper_tree[variable_to_increment]++
+    }
   }
 
   async function get_allowed_files() {
     const route = `${route_prefix}/allowed_files`
+    toggle_loading()
     await api_fetch(
```
[the remark this answers is not in the document]
Comment (JulienChampagnol) [reply components/FileSelector.vue:37]: I just created a new FetchingData component who's  rendering depends on the 'loading' variable

File: components/ExtensionSelector.vue [hunk 8]
```diff
@@ -20,6 +21,8 @@
 </template>
 
 <script setup>
+  import { useToggle } from "@vueuse/core"
```
[the remark this answers is not in the document]
Comment (JulienChampagnol) [reply components/ExtensionSelector.vue:24]: Good idea
